### PR TITLE
Compare Speeds visualization + handler

### PR DESCRIPTION
**Pull request by youralien**
compareSpeeds graphs

comparepercentiles_harvey hardware j 19 00 - 20 00_harvey hardware j 10 00 - 11 00

compare85th_harvey hardware 19 00 - 20 00_harvey hardware 10 00 - 11 00

### CODE REVIEW

**Review by segerphilip — APPROVED** [review]
Graphs for speed comparisons look more understandable, code looks good as well as long as tested 👍

File: app/handlers/baseHandler.py [hunk 1]
```diff
@@ -45,7 +45,9 @@ def find_argument(self, arg_name, expected_type, default=None):
         elif method_type == 'get':
             # Try to get the arg from the header instead
             if expected_type is list:
-                ret_val = self.get_body_arguments(arg_name)
+                print("LIST")
+                ret_val = self.get_arguments(arg_name)
```
Comment (dcelik) [inline app/handlers/baseHandler.py]: Let remove the print statements, but good catch.

File: app/handlers/compareSpeeds.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+import os
+import tornado.web
+from baseHandler import BaseHandler
+from traffic_cloud_utils.app_config import get_project_path, get_project_video_path
+from traffic_cloud_utils.video import get_framerate
+from traffic_cloud_utils.plotting.visualization import vel_distribution, compare_speeds
+
+from traffic_cloud_utils.statusHelper import StatusHelper, Status
+import json
```
Comment (dcelik) [inline app/handlers/compareSpeeds.py]: Do you need the JSON and traceback imports?

File: app/handlers/compareSpeeds.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            if self.only_show_85th:
+                image_filename = 'compare85th_{}.jpg'.format('_'.join(self.labels_to_cmp))
+            else:
+                image_filename = 'comparePercentiles_{}.jpg'.format('_'.join(self.labels_to_cmp))
+            image_path = os.path.join(\
+                                    get_project_path(self.identifier),\
+                                    'final_images',\
+                                    image_filename)
+            self.set_header('Content-Disposition',\
+                            'attachment; filename=velocityPDF.jpg')
```
Comment (dcelik) [inline app/handlers/compareSpeeds.py]: filename should be the image_filename from above

File: app/handlers/compareSpeeds.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            self.set_header('Content-Type', 'application/octet-stream')
+            self.set_header('Content-Description', 'File Transfer')
+            self.write_file_stream(image_path)
+            self.finish("Create Speed Distribution")
```
Comment (dcelik) [inline app/handlers/compareSpeeds.py]: Don't pass back any text in the finish, we should just call finish()

File: app/handlers/createSpeedDistribution.py [hunk 5]
```diff
@@ -29,9 +29,9 @@ def prepare(self):
         self.project_exists(self.identifier)
 
         status_dict = StatusHelper.get_status(self.identifier)
-        if status_dict[Status.Type.SAFETY_ANALYSIS] != Status.Flag.COMPLETE:
+        if status_dict[Status.Type.OBJECT_TRACKING] != Status.Flag.COMPLETE:
```
Comment (dcelik) [inline app/handlers/createSpeedDistribution.py:30]: Need to modify API docs to reflect this change.

**Review by dcelik — CHANGES_REQUESTED** [review]
Looks like a good addition to our visualization codebase.

File: app/handlers/baseHandler.py (hunk 1 above)
Comment (youralien) [reply app/handlers/baseHandler.py]: Removed in following commit

File: app/handlers/createSpeedDistribution.py (hunk 5 above)
Comment (youralien) [reply app/handlers/createSpeedDistribution.py:30]: Changed!

**Review by dcelik — APPROVED** [review]

**Pull request merged**